wallet/src/de/schildbach/wallet/payments/SendCoinsTaskRunner.kt

```diff
@@ -57,12 +61,25 @@ class SendCoinsTaskRunner @Inject constructor(
     private val walletData: WalletDataProvider,
     private val walletApplication: WalletApplication,
     private val securityFunctions: SecurityFunctions,
-    private val packageInfoProvider: PackageInfoProvider
+    private val packageInfoProvider: PackageInfoProvider,
+    coinJoinConfig: CoinJoinConfig
 ) : SendPaymentService {
     companion object {
         private const val WALLET_EXCEPTION_MESSAGE = "this method can't be used before creating the wallet"
         private val log = LoggerFactory.getLogger(SendCoinsTaskRunner::class.java)
     }
+    private var coinJoinSend = false
+    private val coroutineScope = CoroutineScope(Dispatchers.IO)
+
+    init {
+        coinJoinConfig
+            .observeMode()
+            .filterNotNull()
+            .onEach { mode ->
+                coinJoinSend = mode != CoinJoinMode.NONE
+            }
+            .launchIn(coroutineScope)
+    }
```

Comment:
My first thought was to pass a send with coinjoin boolean value in many of these functions.  But this would require that many ViewModels in many modules have access to the `CoinJoinConfig`.

My final idea was to determine the CoinJoin mode here and no other modules or view models need to know anything about CoinJoin.